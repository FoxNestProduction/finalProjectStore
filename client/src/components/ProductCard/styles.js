export const stylesButtonCardOutline = {
  p: '10px',
  borderRadius: '14px',
  color: 'text.primary',
  border: 1,
  fontSize: { mobile: '14px', tablet: '18px' },
  transition: 'background-color 0.3s ease, box-shadow 0.3s ease, color 0.3s ease',
  ':hover': {
    backgroundColor: 'primary.hover',
    boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.3)',
    color: 'text.primaryLight',
  },
};

export const stylesSectionCard = {
  bgcolor: 'background.default',
  boxShadow: 'none',
  borderRadius: 0,
  mb: { mobile: '60px', tablet: '100px' },
};

export const stylesHeaderTopCard = {
  pb: 3,
  pt: 0,
  textAlign: 'center',
  display: { desktop: 'none' },
};

export const stylesHeaderInCard = {
  p: 0,
  textAlign: 'start',
  display: {
    mobile: 'none',
    desktop: 'block',
  },
};

export const stylesMediaCard = {
  mb: 3,
  width: {
    mobile: '70%',
    desktop: '40%',
  },
  maxWidth: {
    tablet: '370px',
  },
  height: 'auto',
  objectFit: 'cover',
};

export const stylesContentCard = {
  display: 'flex',
  flexDirection: {
    mobile: 'column',
    desktop: 'row',
  },
  justifyContent: 'center',
  alignItems: 'center',
  gap: { desktop: '3vw' },
};

export const stylesRatingCard = {
  justifyContent: {
    tablet: 'flex-end',
    desktop: 'flex-start',
  },
  gap: {
    mobile: 1,
    tablet: 7,
    desktop: 16,
  },
  mb: { desktop: 4 },
};

export const stylesLabelCard = {
  flexDirection: { mobile: 'row', desktop: 'column' },
  justifyContent: 'space-between',
  gap: '2px',
};

export const stylesPriceCard = {
  display: 'flex',
  justifyContent: {
    mobile: 'flex-end',
    desktop: 'flex-start',
  },
};

export const stylesButtonCard = {
  p: '10px',
  borderRadius: '14px',
  border: 1,
  fontSize: { mobile: '14px', tablet: '18px' },
  transition: 'background-color 0.3s ease, box-shadow 0.3s ease, color 0.3s ease',
  ':hover': {
    backgroundColor: 'primary.hover',
    boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.3)',
  },
};

export const stylesActionsCard = {
  justifyContent: {
    mobile: 'space-between',
    tablet: 'flex-end',
    desktop: 'space-between',
  },
  gap: {
    mobile: 0,
    tablet: '20px',
    desktop: 0,
  },
  p: 0,
};
