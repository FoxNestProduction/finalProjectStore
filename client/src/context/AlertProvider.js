import React, { createContext, useState, useMemo, memo } from 'react';
import PropTypes from 'prop-types';

export const AlertContext = createContext({});

export const AlertContextProvider = memo(({ children }) => {
  const [alert, setAlert] = useState(false);
  const handleCloseAlert = () => {
    setAlert(false);
  };

  const handleShowAlert = () => {
    setAlert(true);
    setTimeout(() => {
      setAlert(false);
    }, 4000);
  };

  const value = useMemo(() => {
    return {
      alert,
      handleCloseAlert,
      handleShowAlert,
    };
  }, [alert]);

  return <AlertContext.Provider value={value}>{children}</AlertContext.Provider>;
});

AlertContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
