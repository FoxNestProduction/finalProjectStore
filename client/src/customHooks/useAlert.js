import { useContext } from 'react';
import AlertContext from '../context/AlertProvider';

const useAlert = () => {
  return useContext(AlertContext);
};

export default useAlert;
