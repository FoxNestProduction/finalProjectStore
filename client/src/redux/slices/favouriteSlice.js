import { createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { useSelector } from 'react-redux';
import { setUser } from './userSlice';
import { setAuthorizationError } from './errorSlice';

const initialState = {
  favourites: [],
  cardStates: {},
};

const favouriteSlice = createSlice({
  name: 'favourites',
  initialState,
  reducers: {
    setFavourite(state, action) {
      state.favourites = action.payload;
      // localStorage.setItem('favourites', JSON.stringify(state.favourites));
      action.payload.forEach((id) => {
        state.cardStates[id] = true;
      });
    },
    addFavourite(state, action) {
      const { id } = action.payload;
      state.favourites.push(action.payload.id);
      // localStorage.setItem('favourites', JSON.stringify(state.favourites));
      state.cardStates[id] = true;
    },
    removeFavourite(state, action) {
    // eslint-disable-next-line no-underscore-dangle
      const { id } = action.payload;
      const newStateFavourites = state.favourites.filter((item) => item !== action.payload.id);
      state.favourites = newStateFavourites;
      // localStorage.setItem('favourites', JSON.stringify(newStateFavourites));
      state.cardStates[id] = false;
    },
    updateFavourites(state, action) {
      state.favourites = action.payload;
    },
  },
});

export const {
  setIsFavourite,
  addFavourite,
  removeFavourite,
  setFavourite,
  isFavourite,
  updateFavourites,
} = favouriteSlice.actions;

export default favouriteSlice.reducer;
