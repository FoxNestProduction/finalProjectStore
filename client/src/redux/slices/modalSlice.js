import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  isOpen: false,
  title: '',
  content: '',
  buttonBox: false,
  buttonAgree: {
    text: '',
    startIcon: false,
    endIcon: false,
    onClick: null,
  },
};

/* eslint-disable no-param-reassign */

const modalSlice = createSlice({
  name: 'modal',
  initialState,
  reducers: {
    openModal(state) {
      state.isOpen = true;
    },
    closeModal(state) {
      state.isOpen = false;
      state.title = '';
      state.content = '';
      state.buttonBox = false;
      state.buttonAgree.text = '';
      state.buttonAgree.startIcon = false;
      state.buttonAgree.endIcon = false;
      state.buttonAgree.onClick = null;
    },
    setTitle(state, action) {
      state.title = action.payload;
    },
    setContent(state, action) {
      state.content = action.payload;
    },
    setButtonAgree(state, action) {
      state.buttonAgree = action.payload;
    },
    addButtonBox(state) {
      state.buttonBox = true;
    },
  },
});

export const {
  openModal,
  closeModal,
  setTitle,
  setContent,
  setButtonAgree,
  addButtonBox,
} = modalSlice.actions;

/* eslint-enable no-param-reassign */

export default modalSlice.reducer;
