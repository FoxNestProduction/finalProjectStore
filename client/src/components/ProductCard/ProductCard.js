import React, { memo, useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Container from '@mui/material/Container';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import Stack from '@mui/material/Stack';
import Box from '@mui/material/Box';
import Rating from '@mui/material/Rating';
import AddBoxOutlinedIcon from '@mui/icons-material/AddBoxOutlined';
import ColorChips from '../Chip/Chip';
import LoginForm from '../forms/LoginForm/LoginForm';
import FavouriteIcon from '../FavouriteIcon/FavouriteIcon';
import { stylesButtonCard, stylesButtonCardOutline, stylesSectionCard, stylesHeaderTopCard, stylesHeaderInCard, stylesContentCard, stylesActionsCard, stylesPriceCard, stylesRatingCard, stylesLabelCard, stylesMediaCard } from './styles';
import { addToFavourites, deleteFromFavourites, setIsFavourite, removeFavourite } from '../../redux/slices/favouriteSlice';
import { addProductToCart, addToCart, setRestaurants } from '../../redux/slices/cartSlice';
import { openModal, setContent } from '../../redux/slices/modalSlice';
import useAlert from '../../customHooks/useAlert';
import CustomAlert from '../Alert/Alert';

const ProductCard = ({ dish }) => {
  const dispatch = useDispatch();

  const { alert, handleShowAlert, handleCloseAlert } = useAlert();

  const [ishovered, setIsHovered] = useState(false);
  const [isactive, setIsActive] = useState(false);
  const [isShowAlert, setIsShowAlert] = useState(false);

  const isLoading = useSelector((state) => state.favourites.loading);
  const isUserAuthorized = useSelector((state) => state.authorization.isUserAuthorized);

  const {
    name,
    description,
    currentPrice,
    isTrending,
    rating,
    imageUrl,
    isSupreme,
    isHealthy,
    // eslint-disable-next-line no-underscore-dangle
    _id: id,
  } = dish || {};

  const isFavourite = useSelector((state) => state.favourites.cardStates[id]);
  const toggleFavourite = () => {
    if (!isLoading) {
      if (isFavourite) {
        dispatch(removeFavourite(id));
        dispatch(deleteFromFavourites({ id }));
      } else {
        dispatch(setIsFavourite(id));
        dispatch(addToFavourites({ id }));
      }
    }
  };
  const handleOpenModalLogin = () => {
    dispatch(openModal());
    dispatch(setContent(<LoginForm />));
  };

  const handleAddToCart = () => {
    handleShowAlert();
    setIsShowAlert(true);
    setTimeout(() => {
      handleCloseAlert();
      setIsShowAlert(false);
    }, 4000);
    if (isUserAuthorized) {
      dispatch(addProductToCart(id));
    } else {
      const selectedItem = {
        product: { ...dish },
        cartQuantity: 1,
      };
      dispatch(addToCart(selectedItem));
      dispatch(setRestaurants());
    }
  };
  return (
    <Container
      component="section"
      sx={{
        bgcolor: 'background.default',
        mt: { mobile: 5, tablet: 8 },
      }}
    >
      <Card
        sx={stylesSectionCard}
      >
        <CardHeader
          variant="h5"
          component="h1"
          title={name}
          sx={stylesHeaderTopCard}
        />
        <Stack
          sx={stylesContentCard}
        >
          <CardMedia
            component="img"
            image={imageUrl}
            alt="lobster"
            sx={stylesMediaCard}
          />
          <Stack direction="column" sx={{ alignSelf: 'flex-start', width: '100%' }}>
            <CardHeader
              variant="h5"
              component="h3"
              title={name}
              sx={stylesHeaderInCard}
            />
            <Stack
              sx={stylesLabelCard}
            >
              <Box sx={{ my: 2, width: { lgTablet: '350px' } }}>
                <ColorChips
                  isTrending={isTrending}
                  isSupreme={isSupreme}
                  isHealthy={isHealthy}
                />
              </Box>
              <Stack
                direction="row"
                sx={stylesRatingCard}
              >
                <Typography variant="body1" sx={{ whiteSpace: 'nowrap' }}>24min â€¢</Typography>
                <Stack direction="row" spacing={1}>
                  <Rating
                    name="read-only"
                    value={rating ?? 0}
                    size="medium"
                    readOnly
                    sx={{ color: 'primary.main' }}
                  />
                  <Typography component="legend" variant="body1">{rating}</Typography>
                </Stack>
              </Stack>
            </Stack>
            <CardContent sx={{ p: 0, my: 3 }}>
              <Typography
                variant="description"
                component="p"
                sx={{ textAlign: 'justify' }}
              >
                {description}
              </Typography>
            </CardContent>
            <Box
              sx={stylesPriceCard}
            >
              <Typography
                variant="h3"
                sx={{ mb: 3, fontSize: { tablet: '22px', desktop: '30px' } }}
              >
                $
                {currentPrice}
              </Typography>
            </Box>
            <CardActions
              sx={stylesActionsCard}
            >
              <Box
                variant="outlined"
                sx={stylesButtonCardOutline}
                onMouseEnter={() => setIsHovered(true)}
                onMouseLeave={() => setIsHovered(false)}
                onMouseDown={() => setIsActive(true)}
                onMouseUp={() => setIsActive(false)}
                onClick={isUserAuthorized ? toggleFavourite : handleOpenModalLogin}
              >
                Favourite
                <FavouriteIcon id={id} sx={{ ml: 1 }} ishovered={ishovered} isactive={isactive} />
              </Box>
              <Button
                variant="contained"
                sx={stylesButtonCard}
                onClick={handleAddToCart}
              >
                Add to card
                <AddBoxOutlinedIcon
                  fontSize="medium"
                  sx={{ ml: 1 }}
                />
              </Button>
            </CardActions>
          </Stack>
        </Stack>
      </Card>
      {isShowAlert && alert && (
        <CustomAlert type="success" handleCloseAlert={handleCloseAlert} content="Your dish in Cart!" />
      )}
    </Container>
  );
};

ProductCard.propTypes = {
  dish: PropTypes.object,
};

ProductCard.defaultProps = {
  dish: {},
};
export default memo(ProductCard);
