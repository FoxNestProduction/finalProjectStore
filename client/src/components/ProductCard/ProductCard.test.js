import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen, waitFor, act } from '@testing-library/react';
import { AlertContextProvider } from '../../context/AlertProvider';
import useAlert from '../../customHooks/useAlert';
import ProductCard from './ProductCard';
import { addProductToCart, addToCart } from '../../redux/slices/cartSlice';
import { setIsFavourite, addToFavourites, deleteFromFavourites } from '../../redux/slices/favouriteSlice';
import { openModal, setContent } from '../../redux/slices/modalSlice';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('../../customHooks/useAlert', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('../../redux/slices/modalSlice', () => ({
  ...jest.requireActual('../../redux/slices/modalSlice'),
  openModal: jest.fn(),
  setContent: jest.fn(),
}));

jest.mock('../../redux/slices/cartSlice', () => ({
  ...jest.requireActual('../../redux/slices/cartSlice'),
  addProductToCart: jest.fn(),
  addToCart: jest.fn(),
}));

jest.mock('../../redux/slices/favouriteSlice', () => ({
  ...jest.requireActual('../../redux/slices/favouriteSlice'),
  setIsFavourite: jest.fn(),
  addToFavourites: jest.fn(),
  deleteFromFavourites: jest.fn(),
}));

const mockDispatch = jest.spyOn(require('react-redux'), 'useDispatch');

describe('ProductCard Component', () => {

  const dish = {
    name: 'Product',
    description: 'About Product',
    itemNo: '123',
    currentPrice: 10.99,
    isTrending: true,
    rating: 4,
    imageUrl: 'image.jpg',
    isSupreme: false,
    isHealthy: true,
    _id: '123',
  };

  beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);

    useAlert.mockReturnValue({
      alert: false,
      handleShowAlert: jest.fn(),
      handleCloseAlert: jest.fn(),
    });
  });

  test('should renders ProductCard component', () => {
    mockDispatch.mockReturnValueOnce(jest.fn());
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);

    const { asFragment } = render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('should ProductCard added to cart', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);

    render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>,
    );

    const activeButton = screen.getByText('Add to card');

    fireEvent.click(activeButton);

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(addProductToCart).toHaveBeenCalledWith(dish._id);
    expect(useAlert().handleShowAlert).toHaveBeenCalledTimes(1);
    await waitFor(() => {
      expect(useAlert().handleCloseAlert).toHaveBeenCalled();
    }, { timeout: 5000 });
  });

  test('should ProductCard when is not favourite', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);

    render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>
    );

    const favouriteButton = screen.getByText('Favourite');

    fireEvent.click(favouriteButton);

    expect(favouriteButton).toBeInTheDocument();
    expect(dispatch).toHaveBeenCalledTimes(2);
    await waitFor(() => {
      expect(addToFavourites).toHaveBeenCalledWith({ id: '123' });
    });
    expect(setIsFavourite).toHaveBeenCalledWith('123');
  });

  test('renders ProductCard when is favourite', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);

    render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>
    );

    const favouriteButton = screen.getByText('Favourite');

    fireEvent.click(favouriteButton);

    expect(favouriteButton).toBeInTheDocument();
    expect(dispatch).toHaveBeenCalledTimes(2);
    await waitFor(() => {
      expect(deleteFromFavourites).toHaveBeenCalledWith({ id: '123' });
    });
  });

  test('open Modal with LoginForm', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);

    render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>
    );

    const favouriteButton = screen.getByText('Favourite');

    fireEvent.click(favouriteButton);

    expect(dispatch).toHaveBeenCalledTimes(2);
    expect(openModal).toHaveBeenCalled();
    expect(setContent).toHaveBeenCalled();
  });

  test('should add product to cart when user is not authorized', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
  
    render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>
    );
  
    const activeButton = screen.getByText('Add to card');
  
    fireEvent.click(activeButton);
  
    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(addToCart).toHaveBeenCalled();
    expect(useAlert().handleShowAlert).toHaveBeenCalledTimes(1);
    await waitFor(() => {
      expect(useAlert().handleCloseAlert).toHaveBeenCalled();
    }, { timeout: 5000 });
  });

  test('should set isHovered and isActive state on mouse events', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    
    render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>
    );
  
    const favouriteIcon = screen.getByTestId('FavoriteBorderOutlinedIcon');
    fireEvent.mouseEnter(favouriteIcon);
    await waitFor(() => {
      expect(favouriteIcon).toHaveStyle({
        color: 'rgb(156, 39, 176)',
      });
    });
  });

  test('should render rating stars', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    
    render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCard dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>
    );
  
    const ratingStar = screen.getAllByTestId('StarIcon');
    const rating = ratingStar.length;
    const expectedRating = dish.rating ?? 0;
    expect(rating).toBe(expectedRating);
  });
});