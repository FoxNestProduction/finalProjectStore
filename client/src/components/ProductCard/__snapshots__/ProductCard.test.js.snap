// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component should renders ProductCard component 1`] = `
<DocumentFragment>
  <section
    class="MuiContainer-root MuiContainer-maxWidthLg css-2zxudu-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-7ppu4o-MuiPaper-root-MuiCard-root"
    >
      <h1
        class="MuiCardHeader-root css-wft9xy-MuiCardHeader-root"
        variant="h5"
      >
        <div
          class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
          >
            Product
          </span>
        </div>
      </h1>
      <div
        class="MuiStack-root css-hwo7yp-MuiStack-root"
      >
        <img
          alt="lobster"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-meb2d4-MuiCardMedia-root"
          src="image.jpg"
        />
        <div
          class="MuiStack-root css-1am8drf-MuiStack-root"
        >
          <h3
            class="MuiCardHeader-root css-revmsc-MuiCardHeader-root"
            variant="h5"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                Product
              </span>
            </div>
          </h3>
          <div
            class="MuiStack-root css-inz84j-MuiStack-root"
          >
            <div
              class="MuiBox-root css-14p1x4i"
            >
              <div
                class="MuiBox-root css-1ma6dsf"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-u1otj3-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                  >
                    Healthy
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1hn0n2n-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                  >
                    Trending
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiStack-root css-12iqr45-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-rhzpb0-MuiTypography-root"
              >
                24min â€¢
              </p>
              <div
                class="MuiStack-root css-niqf4j-MuiStack-root"
              >
                <span
                  aria-label="4 Stars"
                  class="MuiRating-root MuiRating-sizeMedium Mui-readOnly MuiRating-readOnly css-1ntt3c0-MuiRating-root"
                  role="img"
                >
                  <span>
                    <span
                      class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span>
                    <span
                      class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span>
                    <span
                      class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span>
                    <span
                      class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span>
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <legend
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  4
                </legend>
              </div>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-18m58hg-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-description css-lor93s-MuiTypography-root"
            >
              About Product
            </p>
          </div>
          <div
            class="MuiBox-root css-x39k33"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-rhg1r4-MuiTypography-root"
            >
              $10.99
            </h3>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-us7ea4-MuiCardActions-root"
          >
            <div
              class="MuiBox-root css-18k5jjb"
              variant="outlined"
            >
              Favourite
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-u1vl96-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gick0k-MuiSvgIcon-root"
                  data-testid="FavoriteBorderOutlinedIcon"
                  focusable="false"
                  isactive="false"
                  ishovered="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1oti5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Add to card
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2zunby-MuiSvgIcon-root"
                data-testid="AddBoxOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zm-8-2h2v-4h4v-2h-4V7h-2v4H7v2h4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
